Its.Log.Monitoring/StatefulResult.cs

```diff
@@ -0,0 +1,14 @@
+namespace Its.Log.Monitoring
+{
+    public class StatefulResult<TState, TResult>
+    {
+        public StatefulResult(TState state, TResult result)
+        {
+            State = state;
+            Result = result;
+        }
+
+        public TResult Result { get; }
+        public TState State { get; }
+    }
+}
```

Comment:
im leaning towards 'aggregation' right now... ill make that rename for the time being and we can see if it sticks.